system/jlib/jptree.cpp

```diff
@@ -7881,11 +7879,13 @@ jlib_decl IPropertyTree * loadConfiguration(const char * defaultYaml, const char
         componentDefault.set(defaultConfig->queryPropTree(componentTag));
         if (!componentDefault)
             throw makeStringExceptionV(99, "Default configuration does not contain the tag %s", componentTag);
+        if (*defaultYaml=='{')
+            componentDefault.setown(mapJsonToXmlConfig(componentDefault));
     }
     else
         componentDefault.setown(createPTree(componentTag));
 
-    Owned<IPropertyTree> config(mapJsonToXmlConfig(componentDefault));
```

Comment:
Not sure about this change... I think you are trying to do the mapping if json passed in but not if yaml is, but I would think the mapping is needed in both cases

Reply:
Looks like you did the mapping (#text, lowercase implies attribute) in the yaml reader instead.

Not sure what I think about that - @ghalliday will probably need to comment.

Reply:
We do something similar in JSON.  Although I just realized we don't use "#text" we use "#value".  There are things that we can represent in PTREE and xml that we can't in JSON and YAML.  Supporting this natively allows us to.

XML -> JSON -> YAML -> XML without loosing as much information.
In this case its:
<tag at="1">text or value/<tag>
If fromYAML and toYAML both recognize #value or #text or whatever then that info can be preserved without a user realizing.
Internal XPATH users would  currently realize though, but we could probably add support somehow if we add support complex tag names.

Reply:
The native lower case to attribute mapping would have one side effect:
xml -> yaml -> xml could leave you with more as attributes than the original xml.

Reply:
Really all named scalars could be attributes when reading from Yaml, but:
  - no way to prevent really large content attributes.. how much does that matter?  Doesn't look great serializing to XML.
  - would be even less symmetric when translating back and forth